src/paperoni/__main__.py

```diff
@@ -124,24 +134,118 @@ def run(self):
         self.format(results)
 
 
+@dataclass
+class Work:
+    """Discover and work on prospective papers."""
+
+    @dataclass
+    class Get(Productor):
+        """Get articles from various sources."""
+
+        def run(self, work):
+            focuses = deserialize(Focuses, work.focuses)
+            ex = work.exclusions and deserialize(set[str], work.exclusions)
+            if work.state.exists():
+                top = deserialize(Top[Scored[PaperWorkingSet]], work.state)
+            else:
+                top = Top(work.n)
+            for pinfo in self.iterate():
+                if ex and pinfo.key in ex:
+                    continue
+                scored = Scored(focuses.score(pinfo), PaperWorkingSet.make(pinfo))
+                top.add(scored)
+            dump(Top[Scored[PaperWorkingSet]], top, dest=work.state)
+
+    @dataclass
+    class View:
+        """View the articles in the workset."""
+
+        # Output format
+        format: Formatter = TerminalFormatter
+
+        def run(self, work):
+            top = deserialize(Top[Scored[PaperWorkingSet]], work.state)
+            papers = list(ws.value.current for ws in top)
+            self.format(papers)
+
+    # Command
+    command: TaggedUnion[Get, View]
+
+    # File containing the working set
+    # [alias: -w]
+    state: Path
+
+    # List of focuses
+    # [alias: -f]
+    focuses: Path
```

Comment:
Should focuses be optional here?

Reply:
I don't think it makes much sense not give any.

Reply:
You can try this:

```yaml
- "!institution :: Mila :: 10"
- "!author :: Yoshua Bengio :: 3"
```